addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -653,53 +692,55 @@ registry.mediaVideo = publicWidget.Widget.extend({
-            return def;
-        }
-        var domain = m[1].replace(/^www\./, '');
-        var supportedDomains = ['youtu.be', 'youtube.com', 'youtube-nocookie.com', 'instagram.com', 'vine.co', 'player.vimeo.com', 'vimeo.com', 'dailymotion.com', 'player.youku.com', 'youku.com'];
-        if (!_.contains(supportedDomains, domain)) {
-            // Unsupported domain, don't inject iframe
-            return def;
+        let iframeEl = this.$target[0].querySelector(':scope > iframe');
+        let proms = [this._super.apply(this, arguments)];
+        if (!iframeEl) {
+            // The following code is only there to ensure compatibility with
+            // videos added before bug fixes or new Odoo versions where the
+            // <iframe/> element is properly saved.
+
+            // Bug fix / compatibility: empty the <div/> element as all information
+            // to rebuild the iframe should have been saved on the <div/> element
+            this.$target.empty();
+
+            // Add extra content for size / edition
+            this.$target.append(
+                '<div class="css_editable_mode_display">&nbsp;</div>' +
+                '<div class="media_iframe_video_size">&nbsp;</div>'
+            );
+
+            // Rebuild the iframe. Depending on version / compatibility / instance,
+            // the src is saved in the 'data-src' attribute or the
+            // 'data-oe-expression' one (the latter is used as a workaround in 10.0
+            // system but should obviously be reviewed in master).
+            var src = _.escape(this.$target.data('oe-expression') || this.$target.data('src'));
+            // Validate the src to only accept supported domains we can trust
+            var m = src.match(/^(?:https?:)?\/\/([^/?#]+)/);
+            if (!m) {
+                // Unsupported protocol or wrong URL format, don't inject iframe
+                return Promise.all(proms);
+            }
+            var domain = m[1].replace(/^www\./, '');
+            var supportedDomains = ['youtu.be', 'youtube.com', 'youtube-nocookie.com', 'instagram.com', 'vine.co', 'player.vimeo.com', 'vimeo.com', 'dailymotion.com', 'player.youku.com', 'youku.com'];
+            if (!_.contains(supportedDomains, domain)) {
+                // Unsupported domain, don't inject iframe
+                return Promise.all(proms);
+            }
+            iframeEl = this.$target.append($('<iframe/>', {
+                src: src,
+                frameborder: '0',
+                allowfullscreen: 'allowfullscreen',
+            }))[0];
         }
-        this.$target.append($('<iframe/>', {
-            src: src,
-            frameborder: '0',
-            allowfullscreen: 'allowfullscreen',
-        }));
 
-        return def;
+        proms.push(this._setupAutoplay(iframeEl.getAttribute('src')));
+        return Promise.all(proms).then(() => {
+            this._triggerAutoplay(iframeEl);
+        });
```

Comment:
To not have to reindent the whole code, especially in stable, could not just call a method on the existing iframe and the created iframe? Especially since the iframe creation may be removed at some point in master.

Reply:
Yes it is definitely possible, however there will still be some diffs because of the usage of both `iframeEl` and `proms` by both caller and callee. I'll update the PR :+1: